tests/unit/utils/test_misc.py

```diff
@@ -1,56 +1,37 @@
+import pytest
 from contextlib import contextmanager
 from pathlib import Path
-from tempfile import mkdtemp
-from unittest import TestCase
-
 from ragger.utils import misc
 
 
-class TestMisc(TestCase):
-
-    @contextmanager
-    def directory(self):
-        dir_path = Path(mkdtemp())
-        yield dir_path
-        for filee in dir_path.iterdir():
-            filee.unlink()
-        dir_path.rmdir()
-
-    def test_app_path_from_app_name_ok(self):
-        with self.directory() as dir_path:
-            app_name, device = "some_name", "some_device"
-            expected = (dir_path / f"{app_name}_{device}.elf")
-            expected.touch()
-            self.assertEqual(expected, misc.app_path_from_app_name(dir_path, app_name, device))
+class TestMisc():
 
-    def test_app_path_from_app_name_nok_no_directory(self):
-        with self.assertRaises(AssertionError):
-            misc.app_path_from_app_name(Path("/this/does/not/exists/"), "a", "b")
+    def test_find_project_root_dir_ok(self, root_pytest_dir):
+        assert root_pytest_dir.parent == misc.find_project_root_dir(Path(__file__).parent)
```

Comment:
This is an indirect test. It works because the `root_pytest_dir.parent` happens to have a `.git` directory, which is not guaranteed (by the setup of this test at least). So this test could suddenly fail whereas the tested function were not changed at all. 

You should rather craft a custom FS with a `.git` directory to check this feature.

Reply:
Done